File: app/src/main/java/com/texnar13/deliveryapp/ui/LoginFragment.java
package com.texnar13.deliveryapp.ui;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;

import com.texnar13.deliveryapp.MainViewModel;
import com.texnar13.deliveryapp.MainViewModelFactory;
import com.texnar13.deliveryapp.R;

import java.util.Objects;

public class LoginFragment extends Fragment implements LoginFragmentInterface {

    // обновляемая разметка
    //ProgressBar connectingProgressBar;
    View loginContainer;
    View registerButton;


    static boolean loaded = false;

    // Required empty public constructor
    public LoginFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_login, container, false);

        // скрываемые и показываемые контейнеры
        loginContainer = rootView.findViewById(R.id.fragment_login_input_container);
        registerButton = rootView.findViewById(R.id.fragment_login_go_to_register_button);
        //connectingProgressBar = rootView.findViewById(R.id.fragment_login_load_spinner);
        if (loaded) {
            //connectingProgressBar.setVisibility(View.INVISIBLE);
            loginContainer.setVisibility(View.VISIBLE);
            registerButton.setVisibility(View.VISIBLE);
        } else {
            //connectingProgressBar.setVisibility(View.VISIBLE);
            loginContainer.setVisibility(View.INVISIBLE);
            registerButton.setVisibility(View.INVISIBLE);
        }


        // кнопка авторизации
        rootView.findViewById(R.id.fragment_login_login_button).setOnClickListener(view -> {
            if (loaded) {
                EditText email = rootView.findViewById(R.id.fragment_login_login_edit_email);
                EditText password = rootView.findViewById(R.id.fragment_login_login_edit_password);



                // отдаем данные аутентификации viewModel
                MainViewModel viewModel = (new ViewModelProvider(this,
                        new MainViewModelFactory(requireActivity()))).get(MainViewModel.class);
                viewModel.authUser(
                        email.getText().toString().trim(),
                        password.getText().toString().trim()
                );
            }
        });

        // переход на фрагмент регистрации
        registerButton.setOnClickListener(view -> {
            if (loaded) {

                // переход на страницу регистрации
                Navigation.findNavController(requireActivity(), R.id.activity_main_nav_host_fragment).
                        navigate(R.id.action_loginFragment_to_registerFragment);

            }
        });

        return rootView;
    }


    // Подключение к бд завершено
    @Override
    public void loadOver() {
        Log.e("Hello", "loaded");
        loaded = true;
        //connectingProgressBar.setVisibility(View.INVISIBLE);
        loginContainer.setVisibility(View.VISIBLE);
        registerButton.setVisibility(View.VISIBLE);

//        Toast.makeText(getActivity(), "loaded", Toast.LENGTH_SHORT).show();
    }


}

File: app/src/main/java/com/texnar13/deliveryapp/ui/login/LoginFragment.kt
package com.texnar13.deliveryapp.ui.login

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import com.texnar13.deliveryapp.R
import com.texnar13.deliveryapp.view_model.MainViewModel.Companion.getViewModel


class LoginFragment : Fragment() {

    // обновляемая разметка
    lateinit var loginContainer: View
    lateinit var registerButton: View


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val rootView = inflater.inflate(R.layout.fragment_login, container, false)

        // разметка
        loginContainer = rootView.findViewById(R.id.fragment_login_input_container)
        registerButton = rootView.findViewById(R.id.fragment_login_go_to_register_button)
        val email = rootView.findViewById<EditText>(R.id.fragment_login_login_edit_email)
        val password = rootView.findViewById<EditText>(R.id.fragment_login_login_edit_password)


        // работа с ViewModel
        val viewModel = getViewModel(requireActivity())

        // Загружаем предыдущие вводимые поля
        val lastAuth = viewModel.getUserLastAuth()
        email.setText(lastAuth[0])
        password.setText(lastAuth[1])

        // отслеживаем текущего пользователя
        viewModel.token.observe(viewLifecycleOwner) { token ->
            // если токен был получен и пользователь авторизован
            if (token != null) {
                // переход на страницу пользователя
                findNavController(requireActivity(), R.id.activity_main_nav_host_fragment).navigate(
                        R.id.action_loginFragment_to_userFragment
                )
            }
        }


        // кнопка авторизации
        rootView.findViewById<View>(R.id.fragment_login_login_button).setOnClickListener {


            // отдаем данные аутентификации viewModel
            viewModel.authUser(
                    email.text.toString().trim { it <= ' ' },
                    password.text.toString().trim { it <= ' ' }
            )

        }

        // переход на фрагмент регистрации
        registerButton.setOnClickListener {
            // переход на страницу регистрации
            findNavController(requireActivity(), R.id.activity_main_nav_host_fragment).navigate(
                    R.id.action_loginFragment_to_registerFragment
            )

        }

        return rootView
    }

}